src/core/useAnimations.tsx

```diff
@@ -18,9 +18,12 @@ export function useAnimations<T extends AnimationClip>(
   const [actualRef] = React.useState(() => (root ? (root instanceof Object3D ? { current: root } : root) : ref))
   // eslint-disable-next-line prettier/prettier
   const [mixer] = React.useState(() => new AnimationMixer(undefined as unknown as Object3D))
-  React.useLayoutEffect(() => void ((mixer as any)._root = actualRef.current), [mixer, root])
+  React.useEffect(() => {
```

Comment:
This should still be a `useLayoutEffect` so it's not a frame late. Otherwise LGTM.

Reply:
My understanding of `useLayoutEffect` is that it runs after the component renders (just like `useEffect`), but before the browser paints. And in this case, it's not necessary to have `actualRef.current` and `mixer._root` updated for the browser paint, only for the next time they're referenced as the result of another effect or component render. Since this should be guaranteed by `useEffect` and [it's preferred over `useLayoutEffect` when possible](https://react.dev/reference/react/useLayoutEffect), I think it's the better choice. Let me know if I'm missing something here, and either way, I'm happy to switch it back to `useLayoutEffect` just in case if you'd like.

Reply:
The main thing to consider here is whether this needs to be updated in this effect to ensure a correct result on first render. If that is not relevant or is guaranteed some other way (like via a constructor), then we should keep `useEffect` since React is better able to schedule this as it's async and can be scheduled after many frames whereas `useLayoutEffect` is synchronous and will block work at commit.

Reply:
Right, I think the initial render will behave the same whether we go with `useEffect` or `useLayoutEffect` because they're functionally equivalent in this case (with `useEffect` being marginally better for performance). I'm not sure why `mixer` is being initialized with `new AnimationMixer(undefined)` instead of `new AnimationMixer(<the passed root>)`, but that seems like a separate issue and using `useLayoutEffect` won't set the root properly in time for the first render anyway.

Reply:
You're confusing React render with three.js/browser render. Only `useLayoutEffect` will fire before the latter because of how microtasks are scheduled in the browser. It's not a performance issue as much as whether you want to block those operations, which I'm a little unhappy about the lack of clarification in the docs for.

Reply:
I haven't confused them, I've been referring to React render as "render" and browser render as "paint". Sorry if I wasn't clear. If you're saying this effect should happen by the next browser paint then I totally agree that we need `useLayoutEffect`.

Originally I was thinking that this does not need to happen by the next browser paint because `mixer._root` would not be referenced by anything until the next React render cycle (that is, only the user's React code would be touching `mixer`). But in thinking about it more, I suppose (1) it is quite possible that Three.js would be referencing this before browser paint for some reason, and (2) `requestAnimationFrame` could be scheduled before browser paint and expecting the updated value (e.g. to call `mixer.update`).

Considering all that I think `useLayoutEffect` may be the correct choice, do you agree?

Reply:
I'd stick with `useLayoutEffect` so we don't get any surprises. I'm not worried about perf ramifications in a simple mutation case.